### PR TITLE
feat: added support for -c to set the config file path

### PR DESCRIPTION
**Pull request by vinayaksh42**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
This PR fixes issue #1406 
___
### New feature or improvement
- Describe the details and related test reports.
Added support for "-c" in the managerAPI to set the config File path

### CODE REVIEW

File: api/internal/conf/conf.go [hunk 1]
```diff
@@ -54,6 +54,7 @@ var (
 	SSLDefaultStatus = 1 //enable ssl by default
 	ImportSizeLimit  = 10 * 1024 * 1024
 	PIDPath          = "/tmp/manager-api.pid"
+	FilePathSet		 = "/conf/conf.yaml"
```
Comment (starsz) [inline api/internal/conf/conf.go:57]: Please take care of code style.

**Review by starsz — COMMENTED** [review]
Can you add test in the `/api/test/shell/cli_test.sh`.

File: api/internal/conf/conf.go (hunk 1 above)
Comment (nic-chen) [reply api/internal/conf/conf.go:57]: @vinayaksh42 
You could run `go fmt` to format the code.

Comment (vinayaksh42) [reply api/internal/conf/conf.go:57]: okay i will do that and make another commit to this pr

**Comment by juzhiyuan** [comment]
FE Test failed and waiting for #1426

**Comment by juzhiyuan** [comment]
may sync the latest codes to fix E2E failed.

**Comment by starsz** [comment]
Hi, @vinayaksh42
Maybe you should fix the conflict and sync the master branch.

**Comment by membphis** [comment]
same issue @nic-chen

**Comment by membphis** [comment]
@ vinayaksh42 need to merge the master branch

**Comment by vinayaksh42** [comment]
> @ vinayaksh42 need to merge the master branch

I have merged it.

**Comment by juzhiyuan** [comment]
ping @nic-chen to check.

**Comment by starsz** [comment]
Oh sorry.@vinayaksh42 You need to merge the master branch again.

**Comment by nic-chen** [comment]
need test case for the change

**Comment by vinayaksh42** [comment]
> need test case for the change

Hey can you help me with the modification I need to do in order to add a test case for this?

**Comment by nic-chen** [comment]
@vinayaksh42 
I think we could first refer to the implementation of -c and -p of nginx.

**Comment by starsz** [comment]
Hi @vinayaksh42, do you meet any problem?
I think you should merge the master branch.

**Comment by juzhiyuan** [comment]
> Hi @vinayaksh42, do you meet any problem?
> I think you should merge the master branch.

Could you help to sync codes with this PR?

**Comment by starsz** [comment]
> > Hi @vinayaksh42, do you meet any problem?
> > I think you should merge the master branch.
> 
> Could you help to sync codes with this PR?

Sure, my pleasure.

**Comment by starsz** [comment]
I think we can merge it first.
What do you think? @nic-chen @juzhiyuan @liuxiran @imjoey

**Comment by nic-chen** [comment]
we need some test cases for it.

**Comment by juzhiyuan** [comment]
Hi @vinayaksh42, do you need any help from the community?

**Comment by nic-chen** [comment]
ping @vinayaksh42

**Comment by vinayaksh42** [comment]
> Hi @vinayaksh42, do you need any help from the community?

Hey, I am not sure about how to write the test

**Comment by juzhiyuan** [comment]
This may need @nic-chen @starsz 's help

**Comment by nic-chen** [comment]
Not only need to add tests, but also consider the case where only -c or only -p is specified, or the two parameters are specified together.

**Comment by starsz** [comment]
> > Hi @vinayaksh42, do you need any help from the community?
> 
> Hey, I am not sure about how to write the test

Can you give more details about it? 
So that I can help you solve it.

Maybe you can refer https://github.com/apache/apisix-dashboard/blob/master/api/test/shell/cli_test.sh.

**Comment by LiteSun** [comment]
ping

**Comment by vinayaksh42** [comment]
> ping

Really sorry for the blockage. I won't be able to contribute since I don't have my dev environment (desktop). If it's okay I would like to close this PR?

**Comment by nic-chen** [comment]
> > ping
> 
> Really sorry for the blockage. I won't be able to contribute since I don't have my dev environment (desktop). If it's okay I would like to close this PR?

sure, thanks.

**Pull request closed without merging**